src/seq.rs

```diff
@@ -601,23 +654,41 @@ impl fmt::Debug for Event {
     }
 }
 
-/// Low level methods to set/get data on an Event. Don't use these directly, use generic methods on Event instead.
-pub trait EventData {
+/// Low level method to check if event type supports a given data type. Don't use directly, use generic methods on Event instead.
+pub trait HasEventData {
```

Comment:
> Create a constructor especially for event types with variable length data (e.g. Event::new_varlen) that takes a T: Into<Cow<'a, [u8]>>. That way you can pass either a slice or a vector (by value).

Sounds good to me. As for the naming, I think `new_ext` and `get_ext` could be a good option, as it goes with the `snd_seq_ev_ext_t` type.

 > A corresponding get_data_varlen should return &'a [u8]. What do you think?

`Option<&'a [u8]>` I assume. Do we need a version where you can move the Vec out (called `take_ext` or `into_ext`) without cloning?

:+1: otherwise